turbinia/processors/mount_local.py

```diff
@@ -83,6 +83,86 @@ def GetDiskSize(source_path):
   return size
 
 
+def PreprocessAPFS(source_path, credentials=None):
+  """Uses libfsapfs on a target block device or image file.
+
+  Args:
+    source_path(str): the source path to run fsapfsmount on.
+    credentials(list[(str, str)]): decryption credentials set in evidence setup
+
+  Raises:
+    TurbiniaException: if source_path doesn't exist or if the fsapfsmount
+      command failed to create a virtual device.
+
+  Returns:
+    str: the path to the mounted filesystem.
+  """
+  config.LoadConfig()
+  mount_prefix = config.MOUNT_DIR_PREFIX
+
+  if not os.path.exists(source_path):
+    raise TurbiniaException(
+        'Could not mount partition {0:s}, the path does not exist'.format(
+            source_path))
+
+  if os.path.exists(mount_prefix) and not os.path.isdir(mount_prefix):
+    raise TurbiniaException(
+        'Mount dir {0:s} exists, but is not a directory'.format(mount_prefix))
+  if not os.path.exists(mount_prefix):
+    log.info('Creating local mount parent directory {0:s}'.format(mount_prefix))
+    try:
+      os.makedirs(mount_prefix)
+    except OSError as exception:
+      raise TurbiniaException(
+          'Could not create mount directory {0:s}: {1!s}'.format(
+              mount_prefix, exception))
+
+  mount_path = tempfile.mkdtemp(prefix='turbinia', dir=mount_prefix)
+  mounted = False
+
+  log.debug('Mounting APFS volume')
+
+  if credentials:
+    for credential_type, credential_data in credentials:
+      mount_cmd = ['sudo', 'fsapfsmount']
+      if credential_type == 'password':
+        mount_cmd.extend(['-p', credential_data])
+      elif credential_type == 'recovery_password':
+        mount_cmd.extend(['-r', credential_data])
+      else:
+        # Unsupported credential type, try the next
+        log.warning(
+            'Unsupported credential type: {0!s}'.format(credential_type))
+        continue
+      mount_cmd.extend(['-X', 'allow_other', source_path, mount_path])
+      # Not logging command since it will contain credentials
```

Comment:
Maybe we could still have a log message here, but without the full command line?  Something like "Using fsapfsmount with credential type {...} to mount {foo} on {bar}"?

Reply:
Sure, I also added a similar log message for bdemount.